### PR TITLE
Eren talon subsystem

### PR DESCRIPTION
**Pull request by Ergoold**

### CODE REVIEW

File: src/main/java/com/spikes2212/command/genericsubsystem/TalonSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+public interface TalonSubsystem extends Subsystem {
+    /**
+     * Configure the Talon motor controller and its control loops.
+     */
+    void config();
+
+    /**
+     * Update any control loops running on the Talon.
+     */
+    void update(double setpoint);
```
Comment (ScarletMetal) [inline src/main/java/com/spikes2212/command/genericsubsystem/TalonSubsystem.java]: name this `setSetpoint` or `pidSet`

File: src/main/java/com/spikes2212/command/genericsubsystem/TalonSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/**
+ * A subsystem that runs control loops on a Talon motor controller.
+ *
+ * @author Eran Goldstein
+ */
+public interface TalonSubsystem extends Subsystem {
+    /**
+     * Configure the Talon motor controller and its control loops.
+     */
+    void config();
```
Comment (ScarletMetal) [inline src/main/java/com/spikes2212/command/genericsubsystem/TalonSubsystem.java]: rename this from `config` to `configureLoop`

File: src/main/java/com/spikes2212/command/genericsubsystem/commands/MoveTalonSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     * The {@link TalonSubsystem} this command will run on.
+     */
+    private final TalonSubsystem subsystem;
+
+    /**
+     * The setpoint this command should bring the {@link TalonSubsystem} to.
+     */
+    private final double setpoint;
+
+    public MoveTalonSubsystem(TalonSubsystem subsystem, double setpoint) {
```
Comment (ScarletMetal) [inline src/main/java/com/spikes2212/command/genericsubsystem/commands/MoveTalonSubsystem.java]: make setpoint `Supplier<Double>`

**Comment by Ergoold** [comment]
[Implement TalonSubsystem](https://trello.com/c/D0ib5Lhm/15-implement-talonsubsystem)

File: src/main/java/com/spikes2212/command/genericsubsystem/commands/MoveGenericSubsystemWithPID.java [hunk 4]
```diff
@@ -14,52 +16,65 @@
...
-     * the pid loop that calculates the speed for the command
+     * A supplier that returns the subsystem's current location.
+     */
+    private Supplier<Double> source;
+
+    /**
+     * The PID Settings for the PID control loop.
      */
-    private final PIDLoop pidLoop;
+    private PIDSettings pidSettings;
```
Comment (ScarletMetal) [inline src/main/java/com/spikes2212/command/genericsubsystem/commands/MoveGenericSubsystemWithPID.java]: please add a `FeedFarwardController` instance to this command and adapt the constructor accordingly.

File: src/main/java/com/spikes2212/command/genericsubsystem/commands/MoveGenericSubsystemWithPID.java [hunk 5]
```diff
@@ -14,52 +16,65 @@
...
-    public void initialize() {
-        pidLoop.enable();
+    public MoveGenericSubsystemWithPID(GenericSubsystem subsystem, PIDSettings pidSettings, double setpoint) {
+        this(subsystem, pidSettings, () -> setpoint);
     }
 
     @Override
     public void execute() {
-        pidLoop.update();
+        pidController.setTolerance(pidSettings.getTolerance());
```
Comment (ScarletMetal) [inline src/main/java/com/spikes2212/command/genericsubsystem/commands/MoveGenericSubsystemWithPID.java]: please call `setTolerance` and `setPID` in `initialize`

**Comment by tal-sitton** [comment]
Matan reviewed it

**Pull request merged**